### PR TITLE
Remove variants with no product

### LINKED ISSUE #6828: Deployment of the release draft v3.5.7 failing
## Description

Deployment of the release draft #6818 is failing in staging-FR and -AU with errors, respectively:

staging-FR: https://semaphoreci.com/openfoodfoundation/openfoodnetwork-2/servers/fr-staging/deploys/459
staging-AU: https://semaphoreci.com/openfoodfoundation/openfoodnetwork-2/servers/au-staging/deploys/298

Bugsnag reported:
https://app.bugsnag.com/yaycode/openfoodnetwork/errors/6021138f7ebfc50017fa970f?event_id=6021138f00699cd2ad150000&i=sk&m=nw

Deployment worked for staging-UK. Did not test deploying in staging-ES.

## Steps to Reproduce

Send a deploy job from https://semaphoreci.com/openfoodfoundation/openfoodnetwork-2/branches/master/builds/2151 to staging-FR or -AU

### PR DESCRIPTION
**Pull request by andrewpbrett**
#### What? Why?

Closes #6828 

This migration previously failed if it encountered a variant with a `nil` product. Since this condition causes issues in general, this commit removes those products entirely. 

#### What should we test?

Run this migration with a staging dump of FR and/or AU.

#### Release notes

Changelog Category: Technical changes

### CODE REVIEW

**Comment by andrewpbrett** [comment]
This deployed successfully to FR. Once someone else gives it a double check I think we can merge and unblock the release.

**Review by sauloperez — APPROVED** [review]
I guess there's nothing we can do with those variants :shrug:

**Review by sauloperez — COMMENTED** [review]
mmm do we need any callbacks at all? any reason stopping us from using :rocket: `#destroy_all`?

**Comment by sauloperez** [comment]
thanks @andrewpbrett . That will be definitely faster.

File: db/schema.rb [hunk 1]
```diff
@@ -11,8 +11,7 @@
 #
 # It's strongly recommended that you check this file into your version control system.
 
-
-ActiveRecord::Schema.define(version: 20210203215049) do
```
Comment (sauloperez) [inline db/schema.rb:15]: this former timestamp is really weird. In my machine I have:

```
   up     20210202052337  Migrate variant unit values
   up     20210203214304  Rename migs payment methods to check
```

**Comment by sauloperez** [comment]
Only spec/features/admin/enterprise_roles_spec.rb:132 failed. I feel very uncomfortable doing it but I'll merge it so we don't delay the release any further.

**Pull request merged**